src/net/rpc.rs

```diff
@@ -332,14 +351,147 @@ impl ConversationHttp {
+            clarity_tx.with_clarity_db_readonly(|clarity_db| {
+                let key = ClarityDatabase::make_key_for_account_balance(&account);
+                let (balance, balance_proof) = clarity_db.get_with_proof(&key)
+                    .map(|(a, b)| (a, b.to_hex()))
+                    .unwrap_or_else(|| (0, "".into()));
+                let key = ClarityDatabase::make_key_for_account_nonce(&account);
+                let (nonce, nonce_proof) = clarity_db.get_with_proof(&key)
+                    .map(|(a, b)| (a, b.to_hex()))
+                    .unwrap_or_else(|| (0, "".into()));
+
+                AccountEntryResponse { balance, nonce, balance_proof, nonce_proof }
+            })
+        });
+
+        let response = HttpResponseType::GetAccount(
+            response_metadata, data);
+
+        response.send(http, fd).map(|_| ())
+    }
+
+    fn handle_get_map_entry<W: Write>(http: &mut StacksHttp, fd: &mut W, req: &HttpRequestType,
+                                      chainstate: &mut StacksChainState, cur_burn: &BurnchainHeaderHash, cur_block: &BlockHeaderHash,
+                                      contract_addr: &StacksAddress, contract_name: &ContractName, map_name: &ClarityName, key: &Value) -> Result<(), net_error> {
+        let response_metadata = HttpResponseMetadata::from(req);
+        let contract_identifier = QualifiedContractIdentifier::new(contract_addr.clone().into(), contract_name.clone());
+
+        let data = chainstate.with_read_only_clarity_tx(cur_burn, cur_block, |clarity_tx| {
+            clarity_tx.with_clarity_db_readonly(|clarity_db| {
+                let key = ClarityDatabase::make_key_for_data_map_entry(&contract_identifier, map_name, key);
+                let (value, marf_proof) = clarity_db.get_with_proof::<Value>(&key)
+                    .map(|(a, b)| (a, b.to_hex()))
+                    .unwrap_or_else(|| (Value::none(), "".into()));
+                let data = value.serialize();
+                MapEntryResponse { data, marf_proof }
+            })
+        });
+
+        let response = HttpResponseType::GetMapEntry(
+            response_metadata, data);
+
+        response.send(http, fd).map(|_| ())
+    }
+
+    fn handle_readonly_function_call<W: Write>(http: &mut StacksHttp, fd: &mut W, req: &HttpRequestType,
+                                               chainstate: &mut StacksChainState, cur_burn: &BurnchainHeaderHash,
+                                               cur_block: &BlockHeaderHash, contract_addr: &StacksAddress, contract_name: &ContractName,
+                                               function: &ClarityName, sender: &PrincipalData, args: &[Value] ) -> Result<(), net_error> {
+        let response_metadata = HttpResponseMetadata::from(req);
+        let contract_identifier = QualifiedContractIdentifier::new(contract_addr.clone().into(), contract_name.clone());
+
+        let cost_track = LimitedCostTracker::new(ExecutionCost { write_length: 0, write_count: 0,
```

Comment:
Can you provide some explanation about how these values here are derived?  Also, can we make them configurable and discoverable via `GET /v2/info`?  Rationale is that if I'm querying my own private node, I don't really care too much if I use a ton of resources.  But, if I'm using a public node, then the node operator cares a lot about resource use, and downstream users (namely, their clients) ought to be able to determine the maximum free budgets available to them (so they can report errors early to the user if insufficient resources will lead to application breakage).

Reply:
These are basically just guesses for now. There’s much more work that needs to be done to get reasonable estimates for this (like setting the block limit as well). I agree, though, that this could be configurable.

Reply:
Made configurable via connection options in 5951759a789563a6ecb1ae3800fc0706944b02ec